packages/api/src/module/automation/send-email-when-learning-materials-url-was-generated/learning-materials-url-was-generated-event-handler.service.ts

```diff
@@ -1,12 +1,31 @@
 import { Injectable } from '@nestjs/common';
-import { OnEvent } from '@nestjs/event-emitter';
 
 import { LearningMaterialsUrlWasGenerated } from '@/events/learning-materials-url-was-generated.domain-event';
 import { ApplicationEvent } from '@/module/application-command-events';
+import { EventsSubscription } from '@/write/shared/application/events-subscription/events-subscription';
+import { EventsSubscriptionsRegistry } from '@/write/shared/application/events-subscription/events-subscriptions-registry';
 
 // fixme: example for automation, implement later
 @Injectable()
 export class LearningMaterialsUrlWasGeneratedEventHandler {
-  @OnEvent('LearningMaterialsUrl.*')
-  handleLearningMaterialsUrlDomainEvent(_event: ApplicationEvent<LearningMaterialsUrlWasGenerated>) {}
+  private eventsSubscription: EventsSubscription;
+
+  constructor(private readonly eventsSubscriptionsFactory: EventsSubscriptionsRegistry) {}
+
+  async onModuleInit() {
+    this.eventsSubscription = this.eventsSubscriptionsFactory
+      .subscription('SendEmailWhenLearningMaterialsUrlWasGenerated_Automation_v1')
+      .onEvent<LearningMaterialsUrlWasGenerated>(
+        'LearningMaterialsUrlWasGenerated',
+        this.onLearningMaterialsUrlWasGenerated,
+      )
```

Comment:
I suggest to use nameof() inside onEvent function

https://github.com/dsherret/ts-nameof/blob/master/README.md

Inside onEvent it will be possible to use sth like nameof<DomainEventType>() and it will return given type as string, so here you will use only 

```code
    .onEvent<LearningMaterialsUrlWasGenerated>(this.onLearningMaterialsUrlWasGenerated)
```

Reply:
Very nice idea!
Have you already used it? I have some troubles with configuration. In tests `nameof is not defined`